synedrion/src/paillier/keys.rs

```diff
@@ -1,20 +1,22 @@
+use alloc::boxed::Box;
 use core::fmt::Debug;
 
 use rand_core::CryptoRngCore;
 use serde::{Deserialize, Serialize};
-use zeroize::ZeroizeOnDrop;
+use zeroize::{Zeroize, ZeroizeOnDrop};
 
 use super::params::PaillierParams;
 use crate::uint::{
     subtle::{Choice, ConditionallySelectable},
     Bounded, CheckedAdd, CheckedSub, HasWide, Integer, Invert, NonZero, PowBoundedExp, RandomMod,
     RandomPrimeWithRng, Retrieve, Signed, UintLike, UintModLike,
 };
+use secrecy::{ExposeSecret, SecretBox};
 
-#[derive(Clone, Serialize, Deserialize, ZeroizeOnDrop)]
```

Comment:
I thought if we bound `HalfUint: CloneableSecret + SerializableSecret`, we can derive `Clone` and `Seialize`?

Reply:
It doesn't work (sadly) because the `U512`/`U1024` we use to actually instantiate `HalfUint` do not impl those traits (and both the traits and types are foreign so we can't do it here).